test/screenshot/infra/lib/github-api.js

```diff
@@ -167,16 +167,32 @@ class GitHubApi {
         number: prNumber,
       });
     } catch (err) {
-      throw new VError(err, `Failed to get the base branch for PR #${prNumber}:\n${stackTrace}`);
+      this.error_(`Failed to get the base branch for PR #${prNumber}:`, stackTrace, err);
+      return null;
     }
 
     if (!prResponse.data) {
       const serialized = JSON.stringify(prResponse, null, 2);
-      throw new Error(`Unable to fetch data for GitHub PR #${prNumber}:\n${serialized}`);
+      this.error_(`Unable to fetch data for GitHub PR #${prNumber}:`, serialized);
+      return null;
     }
 
     return `origin/${prResponse.data.base.ref}`;
   }
+
+  /**
+   * @param {string} message
+   * @param {string} stackTrace
+   * @param {(Error|string)=} unsafeErr
+   * @private
+   */
+  error_(message, stackTrace, unsafeErr = '') {
+    let redactedError = nodeUtil.inspect(unsafeErr, /* showHidden */ false, /* depth */ 5);
+    while (this.authToken_ && redactedError.indexOf(this.authToken_) > -1) {
+      redactedError = redactedError.replace(this.authToken_, '[REDACTED]');
```

Comment:
Is there anything else we need to redact? Should we really be spitting out this entire object?

Reply:
I didn't see anything else. I figured it would be better to print the full error message for debugging purposes in case something like this happens again, but I could be persuaded otherwise.

For example, GitHub returns an HTTP 404 for an invalid auth token (instead of HTTP 401/403), and I wouldn't have been able to figure that out had I not been able to see the full error message.

WDYT?

Here's the raw error output:

```
Failed to set commit status::
Error: GitHubApi.setPullRequestStatus()
    at GitHubApi.setPullRequestStatus (/Users/advorak/dev/mdc-web/test/screenshot/infra/lib/github-api.js:84:20)
    at process._tickCallback (internal/process/next_tick.js:68:7)

 { HttpError: Not Found
    at response.text.then.message (/Users/advorak/dev/mdc-web/node_modules/@octokit/request/lib/request.js:55:27)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'HttpError',
  status: 404,
  headers:
   { 'access-control-allow-origin': '*',
     'access-control-expose-headers':
      'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type',
     connection: 'close',
     'content-encoding': 'gzip',
     'content-security-policy': 'default-src \'none\'',
     'content-type': 'application/json; charset=utf-8',
     date: 'Wed, 20 Feb 2019 18:24:28 GMT',
     'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
     server: 'GitHub.com',
     status: '404 Not Found',
     'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
     'transfer-encoding': 'chunked',
     'x-content-type-options': 'nosniff',
     'x-frame-options': 'deny',
     'x-github-media-type': 'github.v3; format=json',
     'x-github-request-id': '8D00:62C7:34C324:813403:5C6D9B5C',
     'x-ratelimit-limit': '60',
     'x-ratelimit-remaining': '57',
     'x-ratelimit-reset': '1550690663',
     'x-xss-protection': '1; mode=block' },
  request:
   { method: 'POST',
     url:
      'https://api.github.com/repos/material-components/material-components-web/statuses/a917eb8ee10d077ff748f4a046da8aab32e7e43a',
     headers:
      { accept: 'application/vnd.github.v3+json',
        'user-agent':
         'octokit.js/16.15.0 Node.js/10.14.2 (macOS High Sierra; x64)',
        authorization: '[REDACTED]',
        'content-type': 'application/json; charset=utf-8' },
     body:
      '{"state":"pending","target_url":"https://github.com/material-components/material-components-web","description":"0 of 1 (0.0%) - 0 diffs","context":"screenshot-test/butter-bot"}',
     request:
      { validate:
         { context: { type: 'string' },
           description: { type: 'string' },
           owner: { required: true, type: 'string' },
           repo: { required: true, type: 'string' },
           sha: { required: true, type: 'string' },
           state:
            { enum: [ 'error', 'failure', 'pending', 'success' ],
              required: true,
              type: 'string' },
           target_url: { type: 'string' } } } },
  documentation_url:
   'https://developer.github.com/v3/repos/statuses/#create-a-status' }
```